lametro/__init__.py

```diff
@@ -16,14 +17,29 @@ class Lametro(Jurisdiction):
         "events": LametroEventScraper,
     }
 
-    legislative_sessions = []
-    for year in range(2014, 2025):
-        session = {
-            "identifier": "{}".format(year),
-            "start_date": "{}-07-01".format(year),
-            "end_date": "{}-06-30".format(year + 1),
-        }
-        legislative_sessions.append(session)
+    @property
+    def legislative_sessions():
+        '''
+        Yield each year that we'd like to scrape today.
+        Allow for the next fiscal year to be scraped during
+        and after the last week of the current fiscal year.
+        '''
+        today = datetime.now()
+        this_year = today.year
+        allowed_years = list(range(2014, this_year))
+
+        if (today.month == 6 and today.day >= 23) or today.month >= 7:
+            # The last week of this fiscal year has begun. Start scraping the next year
+            allowed_years.append(this_year)
+
+        for year in allowed_years:
+            session = {
+                "identifier": "{}".format(year),
+                "start_date": "{}-07-01".format(year),
+                "end_date": "{}-06-30".format(year + 1),
+            }
+            yield session
```

Comment:
`legislative_sessions` should be a list rather than a generator. Since we control its contents and know it will only grow by one each year, we can be sure that the list will be quite small, so we don't need to worry about the performance penalty of creating a list in memory.

Reply:
Got it, I'll get the change in! So generators seem to be more useful for medium to larger datasets and lists are fine when it's a small to medium set of values, does that sound right? Or is this more because we want to preserve the return value of the base class's `legislative_sessions` property?